client.go

```diff
@@ -387,6 +388,9 @@ func (c *Client) Post(dst []byte, url string, postArgs *Args) (statusCode int, b
 // try setting a ReadTimeout.
 func (c *Client) DoTimeout(req *Request, resp *Response, timeout time.Duration) error {
 	req.timeout = timeout
+	if req.timeout < 0 {
```

Comment:
Why not `<=`?

Reply:
`timeout == 0` this is a special case, it is considered that then the timeout is not set  
this behavior is consistent with the standard http driver